src/Queues/Queue.php

```diff
@@ -143,6 +144,32 @@ public function pop(?string $worker = null, ?int $numJobs = null)
         return $numJobs === null ? array_shift($jobs) : $jobs;
     }
 
+    /**
+     * Get job by JID from this queue.
+     *
+     * @param string $jid
+     * @param string|null $worker
+     * @param int|null $numJobs
+     *
+     * @return BaseJob|null
+     */
+    public function popByJid(string $jid, ?string $worker = null, ?int $numJobs = null)
+    {
+        $workerName = $worker ?: $this->client->getWorkerName();
+        $jids = json_decode($this->client->pop($this->name, $workerName, $numJobs ?: 1000), true);
```

Comment:
@osadchyi-s We'll need a Lua code to get exactly one job instead of 1000

Reply:
More Lua methods for God of Lua)

Ok, I agree, it is the more correct decision.